qa-src/Controllers/User/UserPosts.php

```diff
@@ -27,243 +27,329 @@
+			empty($questions) ? 'profile/no_posts_by_x' : 'profile/recent_activity_by_x',
+			$this->userHtml($handle, $profileUserId)
+		);
 
 		// Recent activity by this user
 
-		$qa_content['q_list']['form'] = array(
-			'tags' => 'method="post" action="' . qa_self_html() . '"',
-
-			'hidden' => array(
-				'code' => qa_get_form_security_code('vote'),
-			),
-		);
+		$this->addForm($qa_content);
 
 		$qa_content['q_list']['qs'] = array();
 
-		$htmldefaults = qa_post_html_defaults('Q');
-		$htmldefaults['whoview'] = false;
-		$htmldefaults['voteview'] = false;
-		$htmldefaults['avatarsize'] = 0;
+		$htmlDefaults = qa_post_html_defaults('Q');
+		$htmlDefaults['whoview'] = false;
+		$htmlDefaults['voteview'] = false;
+		$htmlDefaults['avatarsize'] = 0;
 
 		foreach ($questions as $question) {
-			$qa_content['q_list']['qs'][] = qa_any_to_q_html_fields($question, $loginuserid, qa_cookie_get(),
-				$usershtml, null, array('voteview' => false) + qa_post_html_options($question, $htmldefaults));
+			$qa_content['q_list']['qs'][] = qa_any_to_q_html_fields(
+				$question,
+				$loggedInUserId,
+				qa_cookie_get(),
+				$usersHtml,
+				null,
+				array('voteview' => false) + qa_post_html_options($question, $htmlDefaults)
+			);
 		}
 
-
 		// Sub menu for navigation in user pages
 
-		$ismyuser = isset($loginuserid) && $loginuserid == (QA_FINAL_EXTERNAL_USERS ? $this->userid : $useraccount['userid']);
-		$qa_content['navigation']['sub'] = qa_user_sub_navigation($handle, 'activity', $ismyuser);
-
+		$this->addNavigation($qa_content, $handle, $loggedInUserId, $profileUserId, 'activity');
 
 		return $qa_content;
 	}
 
+	/**
+	 * @param string $handle
+	 *
+	 * @return array
+	 * @throws PageNotFoundException
+	 */
 	public function questions($handle)
 	{
-		$this->userHtml($handle);
-
-		$start = qa_get_start();
+		$profileUserId = $this->getUserIdOrFail($handle);
 
 		// Find the questions for this user
 
-		$loginuserid = qa_get_logged_in_userid();
-		$identifier = QA_FINAL_EXTERNAL_USERS ? $this->userid : $handle;
-
-		list($useraccount, $userpoints, $questions) = qa_db_select_with_pending(
-			QA_FINAL_EXTERNAL_USERS ? null : qa_db_user_account_selectspec($handle, false),
-			qa_db_user_points_selectspec($identifier),
-			qa_db_user_recent_qs_selectspec($loginuserid, $identifier, qa_opt_if_loaded('page_size_qs'), $start)
-		);
-
-		if (!QA_FINAL_EXTERNAL_USERS && !is_array($useraccount)) { // check the user exists
-			throw new PageNotFoundException();
-		}
+		$loggedInUserId = qa_get_logged_in_userid();
 
+		list($userPoints, $questions) = $this->fetchQuestionsDataFromDb($handle, $loggedInUserId, $profileUserId);
 
 		// Get information on user questions
 
-		$pagesize = qa_opt('page_size_qs');
-		$count = (int)@$userpoints['qposts'];
-		$questions = array_slice($questions, 0, $pagesize);
-		$usershtml = qa_userids_handles_html($questions, false);
-
+		$pageSize = qa_opt('page_size_qs');
+		$count = (int)@$userPoints['qposts'];
```

Comment:
You should remove `@` char before variable, and `(int)` MUST have space after statement

`$count = (int) $userPoints`